File: java/src/test/java/rocks/iulian/cnp/NifTest.java
package rocks.iulian.cnp;

import org.junit.Test;

import static org.junit.Assert.*;

public class NifTest {

    @Test
    public void shouldBeValid() {
        assertTrue(Nif.isValid("9000123456785"));
    }

    @Test
    public void shouldBeInvalid() {
        assertFalse(Nif.isValid("9000123456780"));
    }

    @Test
    public void shouldConstructWellFormedNif() {
        Nif nif = Nif.fromString("9000123456785");
        assertEquals("9000123456785", nif.stringify());
    }

    @Test
    public void shouldRejectNifWithWrongCheckDigit() {
        Nif nif = Nif.fromString("9000123456780");
        assertNull(nif);
    }

    @Test
    public void shouldRejectIllFormedNif() {
        Nif nif = Nif.fromString("9j91301049995");
        assertNull(nif);
    }

}

File: ruby/test/nif_test.rb
require 'test/unit'
require_relative '../lib/cnp'

class NifTest < Test::Unit::TestCase

  def test_valid
    assert Cnp::Nif.valid?("9000123456785")
  end

  def test_invalid
    assert_false Cnp::Nif.valid?("9000123456780")
  end

  def test_construct_well_formed_nif
    nif = Cnp::Nif.from_string("9000123456785")
    assert_equal "9000123456785", nif.stringify()
  end

  def test_reject_nif_with_wrong_check_digit
    nif = Cnp::Nif.from_string("9000123456780")
    assert_nil nif
  end

  def test_reject_ill_formed_nif
    nif = Cnp::Nif.from_string("9j91301049995")
    assert_nil nif
  end

end